HACKING.md

```diff
@@ -41,6 +35,61 @@ release.
 
 We generally cut a release before disruptive changes land.
 
+### Building on Non-Linux Systems
+
+We provide the `openshift/release` container in which all of our build
+dependencies live, so that one can build a full release of OpenShift without
+having to install anything other than a container runtime on their local system.
+To run scripts or `make` targets from the Origin repo inside of the container,
+use:
+
+    $ hack/env ${COMMAND}
+
+For instance, to build the `oc` binary:
+
+    $ hack/env make build WHAT=cmd/oc
+
+The release container works by streaming a copy of the repository into a volume,
+sharing that volume with the container as it's working directory, executing the
+desired command and streaming the results back to your filesystem.
+
+One can configure what files are uploaded to the container by setting the
+`$OS_BUILD_ENV_EXCLUDE` variable. By default, the entire repository is streamed
+into the container except for anything under `_output/`.
+
+Similarly, to configure what files are downloaded from the container after the
+action is taken, use `$OS_BUILD_ENV_PRESERVE`. By default, `_output/local/bin`,
+`_output/local/releases`, and `_output/scripts` will be downloaded.
+
+### Building Individual Binaries
+
+While `make release` and `make build` will both build all of the binaries required
+for a full release of OpenShift, it is also possible to build individual
+binaries. Binary entrypoints are kept under the `cmd/` directory and can be
+specified to build with the `WHAT` parameter, for instance, to build just `oc`:
+
+    $ make build WHAT=cmd/oc
+
+### Building Individual Images
+
+The `make release` and `hack/build-images.sh` scripts will both build the entire
+suite of images necessary for an OpenShift release. This can take a very long
+time and requires that the full RPM be built before-hand. Individual images
```

Comment:
make release doesn't require the RPM be built before hand, it builds the RPMs right?  i think talking about make release and hack/build-images together is a bit confusing.

you can't run hack/build-images unless you've run make release, and make release builds the images anyway, right?

If there is some shorter chain of commands you can run that allows hack/build-images to work, then we should be explicit about what it is.